### PR TITLE
Allow for unattended installation of extension for CI and development

### PR DESCRIPTION
**Pull request by HDDTHR**
As of right now, anyone wanting to setup directus-schema-sync will have to install it using npx, then restart it to set the `SCHEMA_SYNC` environment variable to start syncing.

I propose that a `SCHEMA_SYNC_AUTO_INSTALL` environment variable be added to automatically install the extension.

I created an example [github repository](https://github.com/HDDTHR/schema-sync-dev-example) demonstrating how this could be used.

### CODE REVIEW

File: src/updateManager.ts [hunk 1]
```diff
@@ -30,28 +30,38 @@ export class UpdateManager {
...
+            hash: newHash,
+            ts: isoTS,
+          };
+          return true;
+        }
+        return false
       }
-
-      return false;
+      // If no row is found (CI), create it
```
Comment (u12206050) [inline src/updateManager.ts:57]: The directus_setting row should be created by Directus, what was the scenario that lead you to needing to create a row?

File: src/updateManager.ts [hunk 2]
```diff
@@ -30,28 +30,38 @@ export class UpdateManager {
 
     const succeeded = await this.db.transaction(async trx => {
       const rows = await trx(this.tableName)
```
Comment (u12206050) [inline src/updateManager.ts:32]: The directus settings is a singleton, so there should just be one row, not sure what this is trying to achieve?

**Review by u12206050 — CHANGES_REQUESTED** [review]
Hey, nice PR.
It looks fairly good, however the changes to updating the columns in the settings table, I am a bit unsure of, mind explaining them, or removing those changes if not really needed.

File: src/updateManager.ts (hunk 1 above)
Comment (HDDTHR) [reply src/updateManager.ts:57]: When initializing Directus for the first time with an empty database, the settings entry won't be created automatically. Instead, it will rely on the default values provided by Directus until the user explicitly saves the settings. I encountered this behavior using the example repository provided above, by debugging the extension and inspecting the database during initialization.

If the directus_settings entry doesn't exist, then this code
https://github.com/bcc-code/directus-schema-sync/blob/1065ee776f35f52f9f1ca1a1c16a72556404a62c/src/updateManager.ts#L27-L59
will return an empty `rows` object, resulting in a true return value for the `lockForUpdates()` method, so the import process won't happen.

https://github.com/bcc-code/directus-schema-sync/blob/1065ee776f35f52f9f1ca1a1c16a72556404a62c/src/index.ts#L89-L91

File: src/updateManager.ts (hunk 2 above)
Comment (HDDTHR) [reply src/updateManager.ts:32]: This is related to the first comment, because the there is a chance that a directus_settings entry won't exist after the initialization of the project.

This code snippet assumes that a directus_settings entry exists and selects it based on the following conditions:
https://github.com/bcc-code/directus-schema-sync/blob/1065ee776f35f52f9f1ca1a1c16a72556404a62c/src/updateManager.ts#L32-L40

If it doesn't exist, then the import process will fail.

Comment (u12206050) [reply src/updateManager.ts:32]: Right, however this would only insert values for the mv_locked/hash/ts but non of the other values. Are you calling `directus install` or `directus bootstrap` at some point to fill those in, and when they are filled in, wouldn't it create a new row?

Comment (HDDTHR) [reply src/updateManager.ts:32]: When launching directus via docker, `cli.js bootstrap` is executed, then `cli.js start` afterwards.

`npx directus bootstrap` and `npx directus install` both initialize the database with tables according to the yaml files in the `directus/api/src/database/seeds` directory. But `npx directus bootstrap` also migrates the database and runs this snippet of code:

https://github.com/directus/directus/blob/13cbe22d07b48f07edcfe85fad4cb8d3a524f12f/api/src/cli/commands/bootstrap/index.ts#L45-L48

As seen above, if the `PROJECT_NAME` environment variable exists, then the `settingsService.upsertSingleton()` method is executed with the appropriate argument. This method updates the `directus_settings` entry if found, otherwise, it will create one. 

I've observed that the created entry contains other values as well, and this due to setting default values in the database table itself as seen by the yaml file below:

https://github.com/directus/directus/blob/13cbe22d07b48f07edcfe85fad4cb8d3a524f12f/api/src/database/seeds/13-settings.yaml

In conclusion, the other values are automatically filled in by the database after creating a new `directus_settings` entry.

**Comment by martin-braun** [comment]
I wonder why this is necessary. I do install `schema-sync` in a deployment scenario too. Your CI/CD pipeline can deploy the files and `npm ci`, `npx directus bootstrap --skipAdminInit` and finally `npx directus schema-sync install` followed by `npx directus schema-sync import`. 

If this is part of your deployment, you don't even need the `SCHEMA_SYNC` env variable and personally I would even refrain from using it. Being explicit about imports and exports gives me peace, knowing that only my own automation can break things, not someone else's.
